Index: src/Models/DBConnect.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/Models/DBConnect.java	(revision 3654659b3b1f6fcb7011426be7a0604a74ca9f3f)
+++ src/Models/DBConnect.java	(date 1537470846076)
@@ -1,5 +1,7 @@
 package Models;
 
+import javafx.collections.ObservableList;
+
 import java.sql.*;
 import java.util.ArrayList;
 
@@ -44,4 +46,131 @@
         }
         return manufacturers;
     }
+
+    public static ArrayList<String> getManufacturersWithOS(String os) throws SQLException {
+        ArrayList<String> manufacturers=new ArrayList<>();
+        Connection conn = null;
+        PreparedStatement ps = null;
+        ResultSet resultSet = null;
+
+        try{
+            //1. connect to the database
+            conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/phones?useSSL=false",
+                    user, password);
+
+            //2.  create a sql statement
+            String sql = "SELECT manufacturer FROM manufacturers WHERE os=?";
+
+            //3.  create the PreparedStatement
+            ps = conn.prepareStatement(sql);
+
+            //4.  bind values to the parameters
+            ps.setString(1, os);
+
+            //5. get the results
+            resultSet = ps.executeQuery();
+
+            //6.  loop over the results from the DB and set the os
+            while (resultSet.next())
+            {
+                manufacturers.add(resultSet.getString("manufacturer"));
+
+            }
+        } catch (Exception e)
+        {
+            System.err.println(e);
+        }
+        finally
+        {
+            if (conn != null)
+                conn.close();
+            if(ps != null)
+                ps.close();
+            if(resultSet != null)
+                resultSet.close();
+        }
+        return manufacturers;
+    }
+
+    public static String getOSWithManufacturer(String manufacturer) throws SQLException {
+        String os=null;
+        Connection conn = null;
+        PreparedStatement ps = null;
+        ResultSet resultSet = null;
+
+        try{
+            //1. connect to the database
+            conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/phones?useSSL=false",
+                    user, password);
+
+            //2.  create a sql statement
+            String sql = "SELECT os FROM manufacturers WHERE manufacturer=?";
+
+            //3.  create the PreparedStatement
+            ps = conn.prepareStatement(sql);
+
+            //4.  bind values to the parameters
+            ps.setString(1, manufacturer);
+
+            //5. get the results
+            resultSet = ps.executeQuery();
+
+            //6.  loop over the results from the DB and set the os
+            while (resultSet.next())
+            {
+                os = resultSet.getString("os");
+            }
+        } catch (Exception e)
+        {
+            System.err.println(e);
+        }
+        finally
+        {
+            if (conn != null)
+                conn.close();
+            if(ps != null)
+                ps.close();
+            if(resultSet != null)
+                resultSet.close();
+        }
+        return os;
+    }
+
+    public static ArrayList<String> getAllOperatingSystems() throws SQLException {
+        ArrayList<String> manufacturers = new ArrayList<>();
+        Connection conn = null;
+        Statement statement = null;
+        ResultSet resultSet = null;
+
+        try{
+            //1. connect to the database
+            conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/phones?useSSL=false",
+                    user, password);
+
+            //2.  create a statement object
+            statement = conn.createStatement();
+
+            //3.  create the SQL query
+            resultSet = statement.executeQuery("SELECT * FROM os");
+
+            //4.  loop over the results from the DB and add to ArrayList
+            while (resultSet.next())
+            {
+                manufacturers.add(resultSet.getString("os"));
+            }
+        } catch (Exception e)
+        {
+            System.err.println(e);
+        }
+        finally
+        {
+            if (conn != null)
+                conn.close();
+            if(statement != null)
+                statement.close();
+            if(resultSet != null)
+                resultSet.close();
+        }
+        return manufacturers;
+    }
 }
Index: src/Models/MobilePhone.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/Models/MobilePhone.java	(revision 3654659b3b1f6fcb7011426be7a0604a74ca9f3f)
+++ src/Models/MobilePhone.java	(date 1537474770019)
@@ -117,4 +117,9 @@
         this.phoneImage = phoneImage;
     }
 
+    public String toString()
+    {
+        return String.format("%s %s with %.0f gigs of memory", this.make,
+                                this.model, this.memory);
+    }
 }
Index: src/Models/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/Models/Main.java	(date 1537453970304)
+++ src/Models/Main.java	(date 1537453970304)
@@ -0,0 +1,29 @@
+package Models;
+
+import javafx.application.Application;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.image.Image;
+import javafx.stage.Stage;
+
+public class Main extends Application {
+    public static void main(String[] args)
+    {
+        //MobilePhone validPhone = new MobilePhone("Apple","iPhone","iOS",10,64,12,12);
+        launch(args);
+    }
+
+    @Override
+    public void start(Stage stage) throws Exception {
+
+        Parent root = FXMLLoader.load(getClass().getResource("../Views/CreatePhoneView.fxml"));
+
+        Scene scene = new Scene(root);
+
+        stage.setScene(scene);
+        stage.setTitle("Create Phone");
+        stage.getIcons().add(new Image("Images/phoneIcon.jpg"));
+        stage.show();
+    }
+}
Index: src/Tests/MobilePhoneTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/Tests/MobilePhoneTest.java	(date 1537387919584)
+++ src/Tests/MobilePhoneTest.java	(date 1537387919584)
@@ -0,0 +1,60 @@
+package Tests;
+
+import Models.MobilePhone;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class MobilePhoneTest {
+
+    MobilePhone validPhone;
+
+    @Before
+    public void setUp() throws Exception {
+        validPhone = new MobilePhone("Apple","iPhone","iOS",10,64,12,12);
+    }
+
+    @Test
+    public void setMake() {
+        try{
+            validPhone.setMake("Fred");
+            fail("Fred is not a valid phone manufacturer");
+        }
+        catch (IllegalArgumentException e){ }
+    }
+
+    @Test
+    public void setModel() {
+        validPhone.setModel("Jaret");
+        assertEquals("Jaret", validPhone.getModel());
+    }
+
+    @Test
+    public void setOs() {
+    }
+
+    @Test
+    public void setScreenSize() {
+    }
+
+    @Test
+    public void setMemory() {
+    }
+
+    @Test
+    public void setFrontCamRes() {
+    }
+
+    @Test
+    public void setRearCamRes() {
+    }
+
+    @Test
+    public void setPrice() {
+    }
+
+    @Test
+    public void setPhoneImage() {
+    }
+}
\ No newline at end of file
Index: src/Views/CreatePhoneView.fxml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/Views/CreatePhoneView.fxml	(revision 3654659b3b1f6fcb7011426be7a0604a74ca9f3f)
+++ src/Views/CreatePhoneView.fxml	(date 1537474842820)
@@ -1,14 +1,143 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
-<?import java.lang.*?>
-<?import java.util.*?>
-<?import javafx.scene.*?>
-<?import javafx.scene.control.*?>
-<?import javafx.scene.layout.*?>
+<?import javafx.geometry.Insets?>
+<?import javafx.scene.control.Button?>
+<?import javafx.scene.control.ChoiceBox?>
+<?import javafx.scene.control.Label?>
+<?import javafx.scene.control.Slider?>
+<?import javafx.scene.control.TextField?>
+<?import javafx.scene.image.ImageView?>
+<?import javafx.scene.layout.AnchorPane?>
+<?import javafx.scene.layout.ColumnConstraints?>
+<?import javafx.scene.layout.GridPane?>
+<?import javafx.scene.layout.HBox?>
+<?import javafx.scene.layout.RowConstraints?>
+<?import javafx.scene.layout.StackPane?>
+<?import javafx.scene.layout.VBox?>
+<?import javafx.scene.text.Font?>
 
-<AnchorPane xmlns="http://javafx.com/javafx"
-            xmlns:fx="http://javafx.com/fxml"
-            fx:controller="Views.CreatePhoneView"
-            prefHeight="400.0" prefWidth="600.0">
-
+<AnchorPane prefHeight="400.0" prefWidth="600.0" xmlns="http://javafx.com/javafx/8.0.121" xmlns:fx="http://javafx.com/fxml/1" fx:controller="Controller.CreatePhoneViewController">
+   <children>
+      <HBox alignment="CENTER" layoutX="29.0" layoutY="65.0" prefHeight="400.0" prefWidth="600.0" spacing="20.0" AnchorPane.bottomAnchor="0.0" AnchorPane.leftAnchor="0.0" AnchorPane.rightAnchor="0.0" AnchorPane.topAnchor="0.0">
+         <children>
+            <GridPane maxWidth="300.0" minWidth="200.0" prefHeight="400.0" prefWidth="276.0">
+              <columnConstraints>
+                <ColumnConstraints hgrow="SOMETIMES" maxWidth="124.79998779296875" minWidth="10.0" prefWidth="118.2000244140625" />
+                <ColumnConstraints hgrow="SOMETIMES" maxWidth="176.0" minWidth="100.0" prefWidth="157.5999755859375" />
+              </columnConstraints>
+              <rowConstraints>
+                <RowConstraints minHeight="10.0" prefHeight="30.0" vgrow="SOMETIMES" />
+                <RowConstraints minHeight="10.0" prefHeight="30.0" vgrow="SOMETIMES" />
+                <RowConstraints minHeight="10.0" prefHeight="30.0" vgrow="SOMETIMES" />
+                  <RowConstraints minHeight="10.0" prefHeight="30.0" vgrow="SOMETIMES" />
+                  <RowConstraints minHeight="10.0" prefHeight="30.0" vgrow="SOMETIMES" />
+                  <RowConstraints minHeight="10.0" prefHeight="30.0" vgrow="SOMETIMES" />
+                  <RowConstraints minHeight="10.0" prefHeight="30.0" vgrow="SOMETIMES" />
+                  <RowConstraints minHeight="10.0" prefHeight="30.0" vgrow="SOMETIMES" />
+              </rowConstraints>
+               <children>
+                  <Label text="Make">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </Label>
+                  <Label text="Model" GridPane.rowIndex="1">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </Label>
+                  <Label text="OS" GridPane.rowIndex="2">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </Label>
+                  <Label text="Screen Size" GridPane.rowIndex="3">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </Label>
+                  <Label text="Memory" GridPane.rowIndex="4">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </Label>
+                  <Label text="Front Camera" GridPane.rowIndex="5">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </Label>
+                  <Label text="Rear Camera" GridPane.rowIndex="6">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </Label>
+                  <Label text="Price" GridPane.rowIndex="7">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </Label>
+                  <ChoiceBox fx:id="makeChoiceBox" prefWidth="175.0" GridPane.columnIndex="1" />
+                  <TextField fx:id="modelTextField" prefWidth="175.0" GridPane.columnIndex="1" GridPane.rowIndex="1">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </TextField>
+                  <ChoiceBox fx:id="osChoiceBox" prefWidth="175.0" GridPane.columnIndex="1" GridPane.rowIndex="2" />
+                  <TextField fx:id="screenSizeTextField" prefWidth="175.0" GridPane.columnIndex="1" GridPane.rowIndex="3">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </TextField>
+                  <HBox alignment="CENTER_LEFT" GridPane.columnIndex="1" GridPane.rowIndex="4">
+                     <children>
+                        <Slider fx:id="memorySlider" onMouseDragged="#updateMemoryLabel" prefHeight="14.0" prefWidth="129.0" />
+                        <Label fx:id="memoryLabel" text="Label" />
+                     </children>
+                  </HBox>
+                  <TextField fx:id="frontCameraTextField" prefWidth="175.0" GridPane.columnIndex="1" GridPane.rowIndex="5">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </TextField>
+                  <TextField fx:id="rearCameraTextField" prefWidth="175.0" GridPane.columnIndex="1" GridPane.rowIndex="6">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </TextField>
+                  <TextField fx:id="priceTextField" prefWidth="175.0" GridPane.columnIndex="1" GridPane.rowIndex="7">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </TextField>
+               </children>
+            </GridPane>
+            <VBox spacing="20.0">
+               <children>
+                  <ImageView fitHeight="179.0" fitWidth="215.0" pickOnBounds="true" preserveRatio="true" />
+                  <StackPane>
+                     <children>
+                        <Button mnemonicParsing="false" prefWidth="215.0" text="Choose Image">
+                           <font>
+                              <Font size="18.0" />
+                           </font>
+                        </Button>
+                     </children>
+                  </StackPane>
+                  <StackPane>
+                     <children>
+                        <Button mnemonicParsing="false" onAction="#createPhoneButtonPushed" prefWidth="215.0" text="Create Phone In System">
+                           <font>
+                              <Font size="18.0" />
+                           </font>
+                        </Button>
+                     </children>
+                  </StackPane>
+               </children>
+            </VBox>
+         </children>
+         <padding>
+            <Insets bottom="20.0" left="20.0" right="20.0" top="20.0" />
+         </padding>
+      </HBox>
+   </children>
 </AnchorPane>
Index: .idea/libraries/mysql_connector_java_5_1_44_bin.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/libraries/mysql_connector_java_5_1_44_bin.xml	(date 1537387566989)
+++ .idea/libraries/mysql_connector_java_5_1_44_bin.xml	(date 1537387566989)
@@ -0,0 +1,9 @@
+<component name="libraryTable">
+  <library name="mysql-connector-java-5.1.44-bin">
+    <CLASSES>
+      <root url="jar://$PROJECT_DIR$/../../NetBeansProjects/mysql-connector-java-5.1.44-bin.jar!/" />
+    </CLASSES>
+    <JAVADOC />
+    <SOURCES />
+  </library>
+</component>
\ No newline at end of file
Index: F18COMP1011W2Prep.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- F18COMP1011W2Prep.iml	(revision 3654659b3b1f6fcb7011426be7a0604a74ca9f3f)
+++ F18COMP1011W2Prep.iml	(date 1537385867380)
@@ -7,5 +7,16 @@
     </content>
     <orderEntry type="inheritedJdk" />
     <orderEntry type="sourceFolder" forTests="false" />
+    <orderEntry type="module-library">
+      <library name="JUnit4">
+        <CLASSES>
+          <root url="jar://$MAVEN_REPOSITORY$/junit/junit/4.12/junit-4.12.jar!/" />
+          <root url="jar://$MAVEN_REPOSITORY$/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar!/" />
+        </CLASSES>
+        <JAVADOC />
+        <SOURCES />
+      </library>
+    </orderEntry>
+    <orderEntry type="library" name="mysql-connector-java-5.1.44-bin" level="project" />
   </component>
 </module>
\ No newline at end of file
